src/Autocomplete/Autocomplete.jsx

```diff
@@ -0,0 +1,205 @@
+import React, {
+  Children,
+  cloneElement,
+  isValidElement,
+  useState,
+  useLayoutEffect,
+  useRef,
+  useMemo
+} from 'react'
+import PropTypes from 'prop-types'
+import styled from 'styled-components'
+import css from '@styled-system/css'
+import { Flex } from '@ivoryio/kogaio'
+
+import Option from '../Dropdown/Option'
+import Heading from '../Dropdown/Heading'
+
+import { Input } from '../Input'
+import { Icon } from '../Icon'
+import sizes from '../assets/sizes'
+
+const Autocomplete = ({
+  disabled,
+  error,
+  label,
+  value,
+  id,
+  searchMaxResults,
+  placeholder,
+  itemSelected,
+  onChange,
+  onSelectItemChange,
+  handleInputChange,
+  tinted,
+  loading,
+  children,
+  ...props
+}) => {
+  const [isCollapsed, setCollapsed] = useState(true)
+  const wrapper = useRef()
+
+  useLayoutEffect(() => {
+    const clickOutside = event => {
+      if (!wrapper.current.contains(event.target)) {
+        setCollapsed(true)
+      }
+    }
+    window.addEventListener('click', clickOutside, { passive: true })
+    return () => {
+      window.removeEventListener('click', clickOutside, { passive: true })
+    }
+  }, [])
+
+  const onSelectItem = option => {
+    if (onSelectItemChange) {
+      onSelectItemChange(option)
+    } else {
+      onChange(null, option)
+    }
+    setTimeout(() => {
+      setCollapsed(true)
+    }, 60)
+  }
+
+  const handleOnChange = event => {
+    onChange(event)
+    handleInputChange && handleInputChange(event.target.value)
+    isCollapsed && setCollapsed(false)
+  }
+
+  const options = useMemo(() => {
+    if (!value) {
+      return Children.toArray(children).slice(0, searchMaxResults)
+    }
+    return Children.toArray(children)
+      .filter(child => child.props.value.indexOf(value.toLowerCase()) > -1)
+      .slice(0, searchMaxResults)
+  }, [value, children, searchMaxResults])
+
+  const isItemSelected = item => (!value ? itemSelected(value, item) : false)
+  const handleInputClick = () => !value && setCollapsed(prev => !prev)
+
+  return (
+    <StyledFlex flexDirection='column' hasLabel={label} width={1} {...props}>
+      <Flex
+        aria-disabled={disabled}
+        aria-haspopup={!disabled}
+        aria-expanded={!isCollapsed}
+        ref={wrapper}
+        alignItems='center'
+        position='relative'
+        width='100%'>
+        <Input
+          error={error}
+          placeholder={placeholder}
+          id={`search-${id}`}
+          type='search'
+          label={label}
+          role='searchbox'
+          disabled={disabled}
+          collapsed={isCollapsed}
+          autocomplete='false'
+          onChange={handleOnChange}
+          value={value}
+          tinted={tinted}
+          showValidationCheck={false}
+          onInputClick={handleInputClick}
+          extension={
+            <>

```

Comment:
is this empty wrapper required? can we just remove it ?

Reply:
No.. it's not required. I will delete it